cmd/crank/main.go

```diff
@@ -18,38 +18,27 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
-
 	"github.com/alecthomas/kong"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	"github.com/crossplane/crossplane-runtime/pkg/logging"
 
 	"github.com/crossplane/crossplane/cmd/crank/beta"
+	v "github.com/crossplane/crossplane/cmd/crank/version"
 	"github.com/crossplane/crossplane/cmd/crank/xpkg"
-	"github.com/crossplane/crossplane/internal/version"
 )
 
 var _ = kong.Must(&cli{})
 
 type (
-	versionFlag string
 	verboseFlag bool
 )
 
 // Decode overrides the default string decoder to be a no-op.
-func (v versionFlag) Decode(_ *kong.DecodeContext) error { return nil }
+func Decode(_ *kong.DecodeContext) error { return nil }
 
 // IsBool indicates that this string flag should be treated as a boolean value.
-func (v versionFlag) IsBool() bool { return true }
-
-// BeforeApply indicates that we want to execute the logic before running any
-// commands.
-func (v versionFlag) BeforeApply(app *kong.Kong) error { //nolint:unparam // BeforeApply requires this signature.
-	fmt.Fprintln(app.Stdout, version.New().GetVersionString())
-	app.Exit(0)
-	return nil
-}
+func IsBool() bool { return true }
```

Comment:
Do `Decode` and `IsBool` still do anything? It looks like they went from being methods on `versionFlag` to being standalone functions.

Reply:
removed